base/download.jl

```diff
@@ -55,50 +9,22 @@ function download_url(url::AbstractString)
     return url
 end
 
-function download(url::AbstractString, filename::AbstractString)
-    url = download_url(url)
-    curl_exe = find_curl()
-    if curl_exe !== nothing
-        return download_curl(curl_exe, url, filename)
-    elseif Sys.iswindows()
-        return download_powershell(url, filename)
-    elseif Sys.which("wget") !== nothing
-        try
-            run(`wget -O $filename $url`)
-        catch
-            rm(filename, force=true)  # wget always creates a file
-            rethrow()
-        end
-    elseif Sys.which("busybox") !== nothing
-        try
-            run(`busybox wget -O $filename $url`)
-        catch
-            rm(filename, force=true)  # wget always creates a file
-            rethrow()
-        end
-    elseif Sys.which("fetch") !== nothing
-        run(`fetch -f $filename $url`)
-    else
-        error("No download agent available; install curl, wget, busybox or fetch.")
-    end
-    return filename
-end
-
-function download(url::AbstractString)
-    filename = tempname()
-    download(url, filename)
-end
+Downloads() = require(PkgId(
```

Comment:
Should `Downloads` just be in Base then (doesn't have to be done now but later)? So far, we've enabled optional features via the presence of standard libraries but now anything that doesn't include the `Downloads` stdlib is arguably a broken build of Julia.

Reply:
It seems better to continue to allow `Downloads` to be a separate body of code which can be used independently. For example, it can be loaded and used on Julia 1.3 as a normal package. If someone is using a Julia build that doesn't have `Downloads` in the sysimg, as long as they don't call `download` everything should be fine. The main improvement here seems like it would be to throw a clearer error if the `Downloads` package isn't a stdlib and hasn't been loaded.

Reply:
I guess I can specialize PackageCompiler to never remove the Downloads stdlib so it is always considered required. It's starting to blur the difference between Julia and the stdlibs a bit uncomfortably though. We already have some of that blur with the type piracy in LinearAlgebra and Random but ideally, it is good to try not to introduce more.

Reply:
I don't really understand the issue. If `Downloads` isn't included but `Base.download` is never called, isn't it fine?